nameko/containers.py

```diff
@@ -127,9 +139,14 @@ def __init__(self, service_cls, worker_ctx_cls, config):
         self.max_workers = (
             config.get(MAX_WORKERS_CONFIG_KEY) or DEFAULT_MAX_WORKERS)
 
-        self.dependencies = DependencySet()
-        for dep in prepare_dependencies(self):
-            self.dependencies.add(dep)
+        # the relationship between extensions and the container is symbiotic;
+        # containers are responsible for starting and stopping their
+        # extensions, and extensions are able to interact with the container
+        # via its :attr:`ServiceContainer.interface`.
+        self.dependencies = ExtensionSet()
```

Comment:
It is a bit weird. If we make the superclass responsible for calling the container lifecycle methods instead then we can definitely refer to the entrypoints and injections/dependencies explicitly.